### DISCUSSION TITLE
How to handle process exit code 1 when polling variable?

### DISCUSSION THREAD

**Discussion by Aiz0** [Aylur/ags#236] [Q&A] [answered]
When a process finishes with exit code 1 it will log an empty error message and I can't find any way to handle that without modifying the command so that the process finishes with exit code 0 instead.

The command false for example always finishes with exit code 1 so heres an example with it,
```js
const alwaysFalse = Variable(0, {
    poll: [1000, 'false', out => 1]
})
```

Can I in js set a default value when the process finishes with exit code 1.

**Comment by Aylur** [comment, marked as answer]
anything other than 0, is considered to be an error, and `Variable` only logs stderr
You could use `Utils.interval` and `Utils.execAsync` and handle the error yourself.
```js
const variable = Variable(0)
interval(1000, () => execAsync('false')
    .then(() => variable.value = 0) // this will never run
    .catch(() => variable.value = 1));
```
